app/src/main/java/edu/uwstout/p2pchat/WifiDirect.java

```diff
@@ -619,4 +618,57 @@ void sendInMemoryFile(final InMemoryFile inMemoryFile)
             Log.e(LOG_TAG, "Attempted to send information before WifiP2pInfo was available.");
         }
     }
+
+    //////////////////////////////////////////////////
+    //
+    // DEPENDENCY INJECTION METHODS AND VARIABLES
+    // Do NOT, for ANY reason, use the methods below
+    // except when performing testing on the WifiDirect
+    // class. In all other cases, these methods are
+    // explicitly not intended to be used.
+    //
+    //////////////////////////////////////////////////
+
+    /**
+     * The IntentService class used to send information
+     * to our peer. When mocked, it does not transmit data.
+     * Default is SendDataService, which is the only acceptable
+     * option during production build.
+     */
+    private Class senderService = SendDataService.class;
```

Comment:
I specifically kept it down there because conceptually that variable only exists to make dependency injection during testing possible, and should not be considered a standard part of the implementation.